### PR TITLE
Give a reasonable image memory size. Every texel size is 4 bytes for now.

### PR DESCRIPTION
**Pull request by locke-lunarg**

### CODE REVIEW

File: icd/generated/mock_icd.cpp [hunk 1]
```diff
@@ -849,6 +856,39 @@ static VKAPI_ATTR VkResult VKAPI_CALL CreateImage(
...
+        case VK_FORMAT_G12X4_B12X4R12X4_2PLANE_420_UNORM_3PACK16:
+        case VK_FORMAT_G12X4_B12X4R12X4_2PLANE_422_UNORM_3PACK16:
+        case VK_FORMAT_G16_B16R16_2PLANE_420_UNORM:
+        case VK_FORMAT_G16_B16R16_2PLANE_422_UNORM:
+            planeCount = 2;
+            break;
+        default:
+            break;
+    }
+    image_memory_size_map[device][*pImage] *= planeCount;
```
Comment (tobine) [inline icd/generated/mock_icd.cpp:891]: Note that you could put this inline for planeCount 2 or 3 above and avoid this extra mul in the common case of planeCount == 1. Saving individual instructions isn't a concern for mock ICD so I'm fine with it this way.

File: icd/generated/mock_icd.cpp [hunk 2]
```diff
@@ -41,6 +41,7 @@ static unordered_map<VkDeviceMemory, VkDeviceSize> allocated_memory_size_map;
 
 static unordered_map<VkDevice, unordered_map<uint32_t, unordered_map<uint32_t, VkQueue>>> queue_map;
 static unordered_map<VkDevice, unordered_map<VkBuffer, VkBufferCreateInfo>> buffer_map;
+static unordered_map<VkDevice, unordered_map<VkImage, VkDeviceSize>> image_memory_size_map;
```
Comment (tobine) [inline icd/generated/mock_icd.cpp:44]: When the device is destroyed we should remove any corresponding entry from this map

Comment (locke-lunarg) [reply icd/generated/mock_icd.cpp:44]: Fixed

File: icd/generated/mock_icd.cpp (hunk 1 above)
Comment (locke-lunarg) [reply icd/generated/mock_icd.cpp]: Fixed

**Pull request merged**